### PR TITLE
Feat/cleaup interface

### PR DESCRIPTION
**Pull request by BelfordZ**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -22,16 +20,15 @@ const cleanBuildDir = async (destinationDirectoryName: string): Promise<any> =>
...
-const compileTemplate = async (name: string, schema: OpenRPC, language: string): Promise<string> => {
-  const methodTypings = new MethodTypings(schema);
+const compileTemplate = async (openrpcDocument: OpenRPC, language: string): Promise<string> => {
+  const methodTypings = new MethodTypings(openrpcDocument);
   await methodTypings.generateTypings();
 
   const template = language === "rust" ? rsTemplate : jsTemplate;
   return template({
-    className: name,
+    className: startCase(openrpcDocument.info.title).replace(/\s/g, ""),
```
Comment (shanejonas) [inline src/index.ts]: you pulled in `trim` but didn't use it here. linting shoulda caught it

File: src/cli.ts [hunk 2]
```diff
@@ -6,18 +6,12 @@ import { parseOpenRPCDocument } from "@open-rpc/schema-utils-js";
 
 program
   .usage("[options] <clientName>")
```
Comment (shanejonas) [inline src/cli.ts]: usage still says `<clientName>` but below you removed it and just used `process.cwd()`

Comment (BelfordZ) [inline src/cli.ts]:
```suggestion
  .usage("[options]")
```

File: src/index.ts [hunk 3]
```diff
@@ -3,16 +3,14 @@ import * as fs from "fs";
 import { ensureDir, emptyDir, copy, move } from "fs-extra";
 import * as path from "path";
 import { promisify } from "util";
-import { map } from "lodash";
+import { map, trim, startCase } from "lodash";
```
Comment (BelfordZ) [inline src/index.ts]:
```suggestion
import { map, startCase } from "lodash";
```

File: src/index.ts (hunk 1 above)
Comment (BelfordZ) [reply src/index.ts]: yeah the linting for 'no unused deps' is busted.

We need to wait for eslint to work for TS. maybe it already does?

**Comment by BelfordZ** [comment]
Blocked by https://github.com/open-rpc/generator-client/issues/169

File: README.md [hunk 4]
```diff
@@ -24,41 +24,29 @@ Supported client languages:
...
-## Usage
-
 ### Generating a Client
 
 ```shell
-open-rpc-generator-client \
-  -s https://raw.githubusercontent.com/open-rpc/examples/master/service-descriptions/petstore-openrpc.json \
-  PetStore
+$ open-rpc-generator-client \
+  -d https://raw.githubusercontent.com/open-rpc/examples/master/service-descriptions/petstore-openrpc.json \
```
Comment (BelfordZ) [inline README.md]:
```suggestion
    -d https://raw.githubusercontent.com/open-rpc/examples/master/service-descriptions/petstore-openrpc.json \
```

**Pull request merged**